src/OpenTelemetry.Exporter.Geneva/GenevaLogExporter.cs

```diff
@@ -242,7 +242,41 @@ internal int SerializeLogRecord(LogRecord logRecord)
         var timestamp = logRecord.Timestamp;
         var cursor = 0;
         cursor = MessagePackSerializer.WriteArrayHeader(buffer, cursor, 3);
-        cursor = MessagePackSerializer.SerializeAsciiString(buffer, cursor, eventName);
+
+        var categoryName = logRecord.CategoryName;
+        string eventName = null;
+
+        Span<byte> tmpEventName = default;
+        int validNameLength = 0;
+
+        // If user configured explicit TableName, use it.
+        if (this.m_tableMappings != null && this.m_tableMappings.TryGetValue(categoryName, out eventName))
+        {
+            cursor = MessagePackSerializer.SerializeAsciiString(buffer, cursor, eventName);
+        }
+        else if (!this.shouldPassThruTableMappings)
+        {
+            eventName = this.m_defaultEventName;
+            cursor = MessagePackSerializer.SerializeAsciiString(buffer, cursor, eventName);
+        }
+        else if (categoryName.Length > 0)
+        {
+            int cursorStartIdx = cursor;
+            cursor = SanitizeCategoryName(buffer, cursor, ref validNameLength, categoryName);
+            if (validNameLength > 0)
+            {
+                tmpEventName = buffer.AsSpan().Slice(cursorStartIdx, validNameLength + 2);
+            }
+            else
+            {
```

Comment:
Add a comment saying: "serializing `null` as categoryName could not be sanitized into a valid string".